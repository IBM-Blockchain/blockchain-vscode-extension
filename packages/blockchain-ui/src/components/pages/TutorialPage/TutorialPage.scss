@import '../../../mixins/layout-mixins.scss';
@import '@carbon/colors/scss/colors';

.tutorial-page-container {
    @include default-page-styling;
    margin-left: $spacing-05;
    margin-top: $spacing-05;
}
