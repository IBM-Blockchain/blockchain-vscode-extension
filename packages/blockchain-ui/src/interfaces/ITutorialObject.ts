interface ITutorialObject {
    title: string;
    series: string;
    length: string;
    firstInSeries?: boolean;
    file: string;
    objectives: string[];
    badge?: boolean;
}

export default ITutorialObject;
