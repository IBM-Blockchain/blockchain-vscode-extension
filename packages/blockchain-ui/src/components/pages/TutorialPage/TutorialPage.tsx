import React, { Component } from 'react';
import HeadingCombo from '../../elements/HeadingCombo/HeadingCombo';
import TutorialTabs from '../../elements/TutorialTabs/TutorialTabs';
import ITutorialObject from '../../../interfaces/ITutorialObject';
import './TutorialPage.scss';

interface IProps {
    tutorialData: Array<{name: string, tutorials: ITutorialObject[]}>;
}

class TutorialPage extends Component<IProps> {
    render(): JSX.Element {
        return (
            <div className='bx--grid tutorial-page-container'>
                <div className='bx--row'>
                    <HeadingCombo
                        headingText='Blockchain Tutorials'
                        subheadingText='Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.'
                    />
                </div>
                <div className='bx--row' id='tutorial-tabs-container'>
                    <TutorialTabs tutorialData={this.props.tutorialData}/>
                </div>
            </div>
        );
    }
}

export default TutorialPage;
