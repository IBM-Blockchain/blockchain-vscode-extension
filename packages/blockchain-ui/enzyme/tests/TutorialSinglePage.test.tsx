import React from 'react';
import renderer from 'react-test-renderer';
import chai from 'chai';
import sinonChai from 'sinon-chai';

import TutorialPage from '../../src/components/pages/TutorialSinglePage/TutorialSinglePage';
import ITutorialObject from '../../src/interfaces/ITutorialObject';

chai.should();
chai.use(sinonChai);

describe('TutorialPage component', () => {

    const tutorial: ITutorialObject = {
      title: 'a2',
      series: 'Basic tutorials',
      length: '4 weeks',
      objectives: [
          'objective 1',
          'objective 2',
          'objective 3'
      ],
      file: 'some/file/path',
      markdown: [
            '## Heading',
            'paragraph',
            '<img src="./relative/link" alt="alt"></img>',
            '<a href="mailto:a link">mailto ignore me</a>',
            '<a href="./resources/package.json">Link to resource file</a>',
            '<a href="./a1.md">Link to another tutorial</a>',
            '<a href="https://ibm.com">Link to external webpage</a>',
            '<a>Bad anchor tag ignore me</a>',
            '<a href="./a1.md"><img src="./image/within/link" alt="alt"></img></a>',
      ].join('\n'),
    };

    const tutorialData: Array<{name: string, tutorials: ITutorialObject[], tutorialFolder: string, tutorialDescription?: string}> = [
        {
            name: 'Basic tutorials',
            tutorialFolder: 'basic-tutorials',
            tutorialDescription: 'some description',
            tutorials: [
              {
                ...tutorial,
                title: 'a1',
                file: './a1.md',
              },
              tutorial,
              {
                ...tutorial,
                title: 'a3',
              },
            ]
        },
        {
            name: 'Other tutorials',
            tutorialFolder: 'other-tutorials',
            tutorials: [
                {
                    title: 'something really interesting',
                    series: 'Other tutorials',
                    length: '10 minutes',
                    objectives: [
                        'objective 1',
                        'objective 2',
                        'objective 3'
                    ],
                    file: 'another/file/path'
                }
            ]
        }
    ];

    it('should render the expected snapshot', () => {
        const component: any = renderer
            .create(<TutorialPage tutorialData={tutorialData} tutorial={tutorial} />)
            .toJSON();
        expect(component).toMatchSnapshot();
    });

    it('should render the component when no tutorialData is passed in', async () => {
        // const component: any = mount(<TutorialPage tutorialData={tutorialData} tutorial={tutorial} />);
        const component: any = renderer
              .create(<TutorialPage tutorialData={[]} tutorial={tutorial} />)
              .toJSON();
        expect(component).toMatchSnapshot();
    });
});
