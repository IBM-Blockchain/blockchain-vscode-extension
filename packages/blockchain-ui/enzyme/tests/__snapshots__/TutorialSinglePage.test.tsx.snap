// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TutorialPage component should render the component when no tutorialData is passed in 1`] = `
<div
  className="bx--grid tutorial-page-container"
>
  <div
    className="bx--row"
  >
    <div
      className="tutorial-page"
    >
      <p>
        <strong>
          IBM Blockchain Platform
        </strong>
      </p>
      <img
        alt="IBM Blockchain Platform"
        className="ibp-icon"
        src="test-file-stub"
      />
      <h2
        className="tutorial-name"
      >
        a2
      </h2>
      <hr />
      <p>
        Estimated time: 
        <code>
          4 weeks
        </code>
      </p>
      <div
        className="tutorial-content"
      >
        <h2>
          Heading
        </h2>
        <p>
          paragraph

          <img
            alt="alt"
            src="../resources/tutorials/some/file/relative/link"
          />
          mailto ignore me
          <a
            className="bx--link"
            href=""
            onClick={[Function]}
          >
            Link to resource file
          </a>
          <a
            href="./a1.md"
          >
            Link to another tutorial
          </a>
          <a
            href="https://ibm.com"
          >
            Link to external webpage
          </a>
          <a>
            Bad anchor tag ignore me
          </a>
          <a
            href="./a1.md"
          >
            <img
              alt="alt"
              src="../resources/tutorials/some/file/image/within/link"
            />
          </a>
        </p>
      </div>
      <hr />
    </div>
  </div>
</div>
`;

exports[`TutorialPage component should render the expected snapshot 1`] = `
<div
  className="bx--grid tutorial-page-container"
>
  <div
    className="bx--row"
  >
    <div
      className="tutorial-page"
    >
      <p>
        <strong>
          IBM Blockchain Platform
        </strong>
      </p>
      <div
        className="tutorial-previous"
      >
        <a
          className="bx--link"
          href=""
          onClick={[Function]}
        >
          ← a1
        </a>
      </div>
      <img
        alt="IBM Blockchain Platform"
        className="ibp-icon"
        src="test-file-stub"
      />
      <h2
        className="tutorial-name"
      >
        a2
      </h2>
      <hr />
      <p>
        Estimated time: 
        <code>
          4 weeks
        </code>
      </p>
      <div
        className="tutorial-content"
      >
        <h2>
          Heading
        </h2>
        <p>
          paragraph

          <img
            alt="alt"
            src="../resources/tutorials/some/file/relative/link"
          />
          mailto ignore me
          <a
            className="bx--link"
            href=""
            onClick={[Function]}
          >
            Link to resource file
          </a>
          <a
            className="bx--link"
            href=""
            onClick={[Function]}
          >
            Link to another tutorial
          </a>
          <a
            href="https://ibm.com"
          >
            Link to external webpage
          </a>
          <a>
            Bad anchor tag ignore me
          </a>
          <a
            className="bx--link"
            href=""
            onClick={[Function]}
          >
            <img
              alt="alt"
              src="../resources/tutorials/some/file/image/within/link"
            />
          </a>
        </p>
      </div>
      <hr />
      <div
        className="tutorial-next"
      >
        <a
          className="bx--link"
          href=""
          onClick={[Function]}
        >
          → a3
        </a>
      </div>
    </div>
  </div>
</div>
`;
